Foundation/FileManager.swift

```diff
@@ -828,11 +828,11 @@ public struct FileAttributeKey : RawRepresentable, Equatable, Hashable {
     public init(rawValue: String) {
         self.rawValue = rawValue
     }
-    
-    public var hashValue: Int {
-        return self.rawValue.hashValue
+
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(rawValue)
```

Comment:
This `hash(into:)` and the `==` below are identical to the definitions that the compiler would synthesize on its own. It may be worth considering removing them here (as well as all other RawRepresentable enums).

Reply:
I agree, esp. given that on Darwin it's the compiler that synthesizes those conformances, right?

Reply:
I looked into it in more detail: on Darwin the hash implementations come from extensions on `_SwiftNewtypeWrapper` that are identical to similar extensions provided on `RawRepresentable`. The stdlib also defines a `==` that’s generic over `RawRepresentable` types; this gets picked up for Equality conformance.

So if we removed these, then the same definitions would be found on `RawRepresentable` instead of compiler synthesis. But the end result is effectively the same; we should do it!